src/amazon/AmazonClient.js

```diff
@@ -29,13 +29,16 @@ class AmazonClient {
   */
   getPairedProducts(walmartProducts, idType='UPC', delayIndex=0) {
     let pairedProducts = new PairedProductList();
-
     return this.getProductsById(walmartProducts.map(item => item.upc), idType, delayIndex)
     .then(function(amazonProducts) {
       amazonProducts.products.forEach(function(amazonProduct) {
         pairedProducts.addPairedProduct(amazonProduct, walmartProducts.find(item => item.upc == amazonProduct.upc));
       });
       return pairedProducts;
+    }).catch(function(error) {
+      console.log(error);
+      // Something went wrong. Return an empty list.
+      return [];
```

Comment:
A function should only have one return type. We are returning an array when there is an error, but returning a `PairedProductList` when success. The `PairedProductList` is very similar to an array, but not the same. It basically serves as a wrapper for a traditional js array in order to provide additional functionality and control. You should return a new `PairedProductList`. Initializing a new one will create an empty array for `pairedProducts.products`.

With all this being said, the return type in the documentation above isn't quite correct so do you mind fixing it? That's my fault.

Reply:
fixed here  59fd1c1